src/WpfMath/UserDefinedColorParser.cs

```diff
@@ -0,0 +1,264 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Windows;
+using System.Windows.Media;
+using WpfMath.Exceptions;
+
+namespace WpfMath
+{
+    /// <summary>
+    /// Parses definitions of colors from a <see cref="string"/>.
+    /// </summary>
+    public class UserDefinedColorParser
+    {
+        /// <summary>
+        /// Parses the <paramref name="input"/> to its <see cref="Color"/> equivalent.
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns></returns>
+        public static Color ParseUserColor(string input)
+        {
+            Color resultColor = new Color();
+            ColorStringTypes CST = GetColorString(input.Trim());
+            switch (CST)
+            {
+                case ColorStringTypes.Byte_longString:
+                    {
+                        byte[] channelbytes = ByteStringValues(input.Trim(), 4);
+                        resultColor.A = channelbytes[0];
+                        resultColor.R = channelbytes[1];
+                        resultColor.G = channelbytes[2];
+                        resultColor.B = channelbytes[3];
+                        break;
+                    }
+                    
+                case ColorStringTypes.Byte_shortString:
+                    {
+                        byte[] channelbytes = ByteStringValues(input.Trim(), 3);
+                        resultColor.A = 255;
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.Hex_longString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 8);
+                        resultColor.A = channelbytes[0];
+                        resultColor.R = channelbytes[1];
+                        resultColor.G = channelbytes[2];
+                        resultColor.B = channelbytes[3];
+                        break;
+                    }
+                case ColorStringTypes.Hex_shortString:
+                    {
+                        byte[] channelbytes = ByteHexStringValues(input.Trim(), 6);
+                        resultColor.A = 255;
+                        resultColor.R = channelbytes[0];
+                        resultColor.G = channelbytes[1];
+                        resultColor.B = channelbytes[2];
+                        break;
+                    }
+                case ColorStringTypes.None:
+                    throw new TexParseException($"The text: {input} could not be parsed.");
+                default:
+                    break;
```

Comment:
I would `throw` in a `default` case, because it's a situation that should never happen.